Engineering/useEffect double calls in React 18.md

```diff
@@ -0,0 +1,78 @@
+---
+tags: engineering, react 18, react hooks, useEffect
+author: T.Hoang Nam (Nigel)
+---
+
+### useEffect double calls in React 18
```

Comment:
I think your note is missing a very important line:

> some effects assume they are only mounted or destroyed once

It means the new behavior might cause trouble for the existing use of useEffect that intends to trigger mount and unmount once. For example:

```js
// This call will be triggered twice and cause an incorrect count in the analytic system
useEffect(() => {
    sendPageVisitAnalytic();
}, [])
```